compiler/z/env/OMRCPU.cpp

```diff
@@ -197,58 +197,11 @@ OMR::Z::CPU::supportsFeatureOldAPI(uint32_t feature)
    }
 
 const char*
-OMR::Z::CPU::getProcessorName(Architecture arch)
-   {
-   const char* result = "";
-   switch (arch)
-      {
-      case Unknown:
-         result = "Unknown";
-         break;
-      case z900:
-         result = "z900";
-         break;
-      case z990:
-         result = "z990";
-         break;
-      case z9:
-         result = "z9";
-         break;
-      case z10:
-         result = "z10";
-         break;
-      case z196:
-         result = "z196";
-         break;
-      case zEC12:
-         result = "zEC12";
-         break;
-      case z13:
-         result = "z13";
-         break;
-      case z14:
-         result = "z14";
-         break;
-      case z15:
-         result = "z15";
-         break;
-      case zNext:
-         result = "zNext";
-         break;
-      default:
-         TR_ASSERT(false, "Invalid Archiecture Type: %d", arch);
-         result = "Invalid Architecture type!";
-         break;
-      }
-   return result;
-   }
-
-const char*
-OMR::Z::CPU::getProcessorName(OMRProcessorArchitecture arch)
+OMR::Z::CPU::getProcessorName()
```

Comment:
I think we need a bit more work here. We now have duplication between OMR/OpenJ9. See:

https://github.com/eclipse-openj9/openj9/blob/a82dc74872dc55d92d90b7af4751bdfa0980b0f9/runtime/compiler/env/ProcessorDetection.cpp#L294-L542

We need to eliminate this duplication while we're in the area here. Do you want to tackle this here or open another issue to remove this? IMO `getProcessorName` being reviewed here should be in the common CPU class which just returns "Unknown" and then we override on each platform via extensible classes like you do here.

Reply:
> IMO getProcessorName being reviewed here should be in the common CPU class which just returns "Unknown" and then we override on each platform via extensible classes like you do here

Can't we do all this work for all the CPUs in the common CPU class? If so then isn't it better to do it there?

Reply:
If you do it at the common level you'll have to have a bunch of `if (isZ)`, `else if (isPower)`, etc. which is not really what we want.

Reply:
Changes made in 03905d340e70e72dd1c91acc45dd06949c47c9dc. This PR is ready for another review.